src/Microsoft.AspNet.Server.Kestrel/Infrastructure/KestrelTrace.cs

```diff
@@ -12,6 +12,32 @@ namespace Microsoft.AspNet.Server.Kestrel
     /// </summary>
     public class KestrelTrace : IKestrelTrace
     {
+        static Action<ILogger, long, Exception> _connectionStart;
+        static Action<ILogger, long, Exception> _connectionStop;
+        static Action<ILogger, long, Exception> _connectionPause;
+        static Action<ILogger, long, Exception> _connectionResume;
+        static Action<ILogger, long, Exception> _connectionReadFin;
+        static Action<ILogger, long, Exception> _connectionWriteFin;
+        static Action<ILogger, long, int, Exception> _connectionWroteFin;
+        static Action<ILogger, long, Exception> _connectionKeepAlive;
+        static Action<ILogger, long, Exception> _connectionDisconnect;
+
+        static KestrelTrace()
+        {
+            _connectionStart = LoggerMessage.Define<long>(LogLevel.Debug, 1, @"Connection id ""{ConnectionId}"" started.");
+            _connectionStop = LoggerMessage.Define<long>(LogLevel.Debug, 2, @"Connection id ""{ConnectionId}"" stopped.");
+            // ConnectionRead 3
+            _connectionPause = LoggerMessage.Define<long>(LogLevel.Debug, 4, @"Connection id ""{ConnectionId}"" paused.");
+            _connectionResume = LoggerMessage.Define<long>(LogLevel.Debug, 5, @"Connection id ""{ConnectionId}"" resumed.");
+            _connectionReadFin = LoggerMessage.Define<long>(LogLevel.Debug, 6, @"Connection id ""{ConnectionId}"" received FIN.");
+            _connectionWriteFin = LoggerMessage.Define<long>(LogLevel.Debug, 7, @"Connection id ""{ConnectionId}"" sending FIN.");
+            _connectionWroteFin = LoggerMessage.Define<long, int>(LogLevel.Debug, 8, @"Connection id ""{ConnectionId}"" sent FIN with status ""{Status}"".");
+            _connectionKeepAlive = LoggerMessage.Define<long>(LogLevel.Debug, 9, @"Connection id ""{ConnectionId}"" completed keep alive response.");
+            _connectionDisconnect = LoggerMessage.Define<long>(LogLevel.Error, 10, @"Connection id ""{ConnectionId}"" disconnected.");
```

Comment:
I'd imagine so; but would need the following overload (as `Exception` is is a built in param)?

```csharp
static Action<ILogger, Exception> Define(LogLevel logLevel, int eventId, string formatString)
```
in `Microsoft.Extensions.Logging.Abstractions/LoggerMessage.cs` which doesn't currently exist.